src/core/core-varlink.c

```diff
@@ -202,11 +203,23 @@ static int vl_method_subscribe_managed_oom_cgroups(
                 void *userdata) {
 
         _cleanup_(json_variant_unrefp) JsonVariant *v = NULL;
+        _cleanup_free_ char *comm = NULL;
         Manager *m = ASSERT_PTR(userdata);
+        pid_t pid;
         int r;
 
         assert(link);
 
+        r = varlink_get_peer_pid(link, &pid);
+        if (r < 0)
+                return r;
+
+        (void) get_process_comm(pid, &comm);
+
+        /* This is meant to be a deterrent and not actual security. */
+        if (!streq(comm, "systemd-oomd"))
+                return varlink_error(link, VARLINK_ERROR_PERMISSION_DENIED, NULL);
```

Comment:
hmm, why not do actual security here? i.e. systemd-oomd already runs as its own user `systemd-oom`, so we could check the sender's UID and verify that it matches the UID of the `systemd-oom` user?

Reply:
This was what I did in the first version of this PR. However @keszybz brought up that NSS lookups are blocking and that there is a policy against doing them pid1.

Reply:
checking that the pid is in the systemd-oomd service should be good enough?

Reply:
ah the nss issue is indeed a problem. Would be good to add a comment explaining why UID is not used there.